### DISCUSSION TITLE
How to set these condition

### DISCUSSION THREAD

**Discussion by icnosvbovwb** [OpenFAST/openfast#1854] [Q&A]
Dear @jjonkman 
I want to simulate idling condition, parked condition and emergency condition with ITI barge wind turbine，please advice me on whether it is correct or not to setup.
1. idling condition
NRELOffshrBsline5MW_ITIBarge4_ServoDyn.dat
![image](https://github.com/OpenFAST/openfast/assets/107483730/9cde4670-c733-4c63-81ab-e6cb55b185dd)

NRELOffshrBsline5MW_ITIBarge4_ElastoDyn.dat
![image](https://github.com/OpenFAST/openfast/assets/107483730/2fa5cc54-3fbf-42ed-89df-4873b9bd76a8)

NRELOffshrBsline5MW_AeroDyn_Equil_noTwr.dat
![image](https://github.com/OpenFAST/openfast/assets/107483730/1c86b5ad-ed84-4131-bde0-ffc1600d85eb)
Do these pictures look correct？
2. parked condition
Do these two conditions（ idling condition and parked condition） represent the same?

3.emergency condition
Assuming the total time is 1000s, I need to run 500s and stop. Is the following setting correct?
NRELOffshrBsline5MW_ITIBarge4_ElastoDyn.dat
![image](https://github.com/OpenFAST/openfast/assets/107483730/3c927acb-bbf6-4317-b388-47ad50590af7)

NRELOffshrBsline5MW_ITIBarge4_ServoDyn.dat
![image](https://github.com/OpenFAST/openfast/assets/107483730/18f2be8f-5f6d-4e30-8811-993de13a8f9b)

**Comment by jjonkman** [comment]
Dear @icnosvbovwb,

Overall your input file settings make sense to me.  Just a few comments:
-  "Parked" typically means with the generator locked.  You can model this (without consideration of slippage of the shaft brake) by disabling the generator DOF in ElastoDyn (`GenDOF` = False).
-  I'm not fully sure I know you mean by "emergency", but it looks like you are overriding the pitch controller by feathering the blades to 90 degrees, starting at 500 s.  I would normally expect such an event to occur in conjunction with a shut down of the generator, so, I would expect that you'd also set `TimGenOf` in ServoDyn to be around 500 s for this case.
-  For the cases with active pitch and torque control, it look like you are specifying the controllers in Simulink, which you haven't shared, so, I can't comment on those.

Best regards,

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
Q1:I want to set parked condition. Is this setting correct?

PCMode = 0 (disable pitch control)
GenTiStr = True
TimGenOn = 9999.9 (a large don’t care > TMax)
BlPitch1 = BlPitch2 = BlPitch3 = 90.0 (fully feathered)
RotSpeed = 0.0 (or close to zero)
StallMod = STEADY (disable dynamic stall, which isn’t valid at large angles of attack)
IndModel = NONE (disable aerodynamic induction)
GenDOF = False

Q2: The idling setting includes this (TimGenOn = 9999.9). I am not sure what the meaning of this item is. It takes 10000 seconds to start the generator, but the actual total calculation is 1000 seconds, and the genspeed will have a positive or negative value of 10rpm. Is this normal? If (GenDOF = False) is added to the idling condition, the genspeed will always be 0. So my understanding is that the generator should be 0 in idling condition, why is there a positive or negative value of 10rpm under the condition of (TimGenOn = 9999.9). Shouldn't it wait until 10000 seconds for the generator to have data?
![image](https://github.com/OpenFAST/openfast/assets/107483730/89f21931-3d0a-4d2b-8125-35480f729d39)
![image](https://github.com/OpenFAST/openfast/assets/107483730/cb86eaab-d2f1-421a-bbb2-ccaba5be98ec)
![image](https://github.com/OpenFAST/openfast/assets/107483730/182abc7a-7019-403c-959e-19c3782582f2)

Q3: The emergency event is when the cabin acceleration exceeds 2m/s/s, it will initiate a shutdown and 90 ° feathering. Where should I set this condition? Or how can I modify the source code?

**Reply by jjonkman** [reply]
Dear @icnosvbovwb,

Here are my responses:

1. Your settings look fine.  Please note, though, that inputs `GenTiStr` and `TimGenOn` in ServoDyn wont have any effect when `GenDOF` = FALSE in ElastoDyn.
2. `TimGenOn` is the time the generator will kick-in when `GenTiStr` = TRUE.  If `TimGenOn` > `TMax` (the total simulation time), then the generator will remain off for the entire simulation, which is what you want to model an idling rotor.  Even if the generator is offline, the rotor/generator can still move around a bit due to the applied aerodynamic loads on the rotor (when `GenDOF` = False).
3. The simple shutdown options available through the ServoDyn input file do not support a shutdown triggered by nacelle-acceleration exceeding some threshold.  But you could always implement your own safety and protection system logic in your own user-defined torque and pitch controller.

Best regards,
